clients/src/main/java/org/apache/kafka/common/requests/ConsumerGroupHeartbeatRequest.java

```diff
@@ -59,6 +60,9 @@ public Builder(ConsumerGroupHeartbeatRequestData data, boolean enableUnstableLas
 
         @Override
         public ConsumerGroupHeartbeatRequest build(short version) {
+            if (version == 0 && data.subscribedTopicRegex() != null) {
+                throw new UnsupportedVersionException("Broker does not support resolution of SubscriptionPattern on version " + version);
```

Comment:
you're right, we have the same exception but in 2 diff cases, so I made the changes to separate the messages, one for the case of HB API not supported (protocol is not enabled), the other one for when a required HB version is not supported (ex. v1 for regex). Also changed the msgs a bit to make them consistent, thoughts?

Reply:
Pardon me, why do we need to handle the version check manually instead of relying on the generated code? Alternatively, is this PR intended to add more meaningful error messages? If so, could you please include the explicit valid version number (or broker version) in the error message?

Reply:
We could indeed rely on the auto-generated code but the error message is not so great.

Reply:
Agree, this allows us to show an error msg specific to the field and version required (and it's also detecting it while building the request, so avoiding the actual HB request doomed to fail). 

I updated the msg to add the version required as suggested.